OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -2720,8 +2723,11 @@ public static DreamValue NativeProc_sorttext(NativeProc.State state)
         [DreamProcParameter("n", Type = DreamValueType.Float)]
         public static DreamValue NativeProc_trunc(NativeProc.State state) {
             DreamValue arg = state.GetArgument(0, "n");
-            if (arg.TryGetValueAsFloat(out float floatnum)) {
-                return new DreamValue(MathF.Truncate(floatnum));
+            if (arg.TryGetValueAsFloat(out var floatNum)) {
+                if (!float.IsFinite(floatNum)) {
+                    return arg;
+                }
+                return new DreamValue((int)floatNum);
```

Comment:
Does this not just give the same values as `MathF.Truncate()`?

Reply:
Yes, but imo in this case its cleaner since its just turning it into a integer

Reply:
I don't see how this is cleaner and preferable over a one-line call to the `MathF` equivalent of the proc.